lib/experimental/class-wp-rest-post-archive-search-handler.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * REST API: WP_REST_Post_Archive_Search_Handler class
+ *
+ * @package WordPress
+ * @subpackage REST_API
+ * @since 6.6.0
+ */
+
+/**
+ * Core class representing a search handler for Post Archives in the REST API.
+ *
+ * @since 5.6.0
+ *
+ * @see WP_REST_Search_Handler
+ */
+class WP_REST_Post_Archive_Search_Handler extends WP_REST_Search_Handler {
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 5.6.0
+	 */
+	public function __construct() {
+		$this->type = 'post-type-archive';
+	}
+
+	/**
+	 * Searches post-type archives for a given search request.
+	 *
+	 * @since 5.6.0
+	 *
+	 * @param WP_REST_Request $request Full REST request.
+	 * @return array {
+	 *     Associative array containing found IDs and total count for the matching search results.
+	 *
+	 *     @type int[]               $ids   Found post archive IDs.
+	 *     @type string|int|WP_Error $total Numeric string containing the number of post-type archives found, or WP_Error object.
+	 * }
+	 */
+	public function search_items( WP_REST_Request $request ) {
```

Comment:
Need to check possible `$request` params for the search handler endpoints and ensure we're handling those correctly.

For example, offset and pagination...etc.

Reply:
Here are the [Search endpoint docs](https://developer.wordpress.org/rest-api/reference/search-results/).

We're handling pagination based on [this example from post-formats](https://github.com/WordPress/wordpress-develop/blob/8f42bad4cdcb5fb6c4ff96fcf050fb29ff6d30bf/src/wp-includes/rest-api/search/class-wp-rest-post-format-search-handler.php#L84).